Keyboards/LanguageKeyboards/Hebrew/HEInterfaceVariables.swift

```diff
@@ -67,70 +48,158 @@ struct HebrewKeyboardProvider: KeyboardProviderProtocol {
-      .addRow([SpecialKeys.capsLock, "°", "/", ":", ";", "(", ")", "$", "&", "@", "£", "¥", "€", "return"]) // "undo"
-      .addRow(["shift", "…", "?", "!", "~", "≠", "'", "\"", "_", ",", ".", "-", "shift"]) // "redo"
-      .addRow(["selectKeyboard", "ABC", "space", "ABC", "hideKeyboard"]) // "microphone", "scribble"
-      .build()
+/// Gets the keys for the Hebrew keyboard.
+
+
+func getHEKeys() {
+  guard let userDefaults = UserDefaults(suiteName: "group.be.scri.userDefaultsContainer") else {
+    fatalError()
+  }
+  
+  var currencyKey = HebrewKeyboardConstants.defaultCurrencyKey
+  var currencyKeys = HebrewKeyboardConstants.currencyKeys
+  let dictionaryKey = controllerLanguage + "defaultCurrencySymbol"
+  
+  if let currencyValue = userDefaults.string(forKey: dictionaryKey) {
+    currencyKey = currencyValue
+  } else {
+    userDefaults.setValue(currencyKey, forKey: dictionaryKey)
+  }
+  if let index = currencyKeys.firstIndex(of: currencyKey) {
+    currencyKeys.remove(at: index)
+  }
+  
+  if DeviceType.isPhone {
+    letterKeys = HebrewKeyboardProvider.genPhoneLetterKeys()
+    numberKeys = HebrewKeyboardProvider.genPhoneNumberKeys(currencyKey: currencyKey)
+    symbolKeys = HebrewKeyboardProvider.genPhoneSymbolKeys(currencyKeys: currencyKeys)
+    allKeys = Array(letterKeys.joined()) + Array(numberKeys.joined()) + Array(symbolKeys.joined())
+    
+    leftKeyChars = ["ק", "1", "-", "[", "_"]
+    rightKeyChars = ["פ", "0", "\"", "=", "·"]
+    centralKeyChars = allKeys.filter { !leftKeyChars.contains($0) && !rightKeyChars.contains($0) }
+  } else {
+    // Use the expanded keys layout if the iPad is wide enough and has no home button.
+    if usingExpandedKeyboard {
+      letterKeys = HebrewKeyboardProvider.genPadExpandedLetterKeys()
+      symbolKeys = HebrewKeyboardProvider.genPadExpandedSymbolKeys()
+      
+      allKeys = Array(letterKeys.joined()) + Array(symbolKeys.joined())
+    } else {
+      letterKeys = HebrewKeyboardProvider.genPadLetterKeys()
+      numberKeys = HebrewKeyboardProvider.genPadNumberKeys(currencyKey: currencyKey)
+      symbolKeys = HebrewKeyboardProvider.genPadSymbolKeys(currencyKeys: currencyKeys)
+      
+      letterKeys.removeFirst(1)
+      
+      allKeys = Array(letterKeys.joined()) + Array(numberKeys.joined()) + Array(symbolKeys.joined())
+    }
+    
+    leftKeyChars = ["1", "ק"]
+    rightKeyChars = ["הכנס", "hideKeyboard"]
+    centralKeyChars = allKeys.filter { !leftKeyChars.contains($0) && !rightKeyChars.contains($0) }
   }
 }
 
-/// Gets the keys for the Hebrew keyboard.
-func getHEKeys() {}
 
-/// Provides the Hebrew keyboard layout.
-func setHEKeyboardLayout() {}
+func setHEKeyboardLayout() {
+  getHEKeys()
+
+  currencySymbol = ""
+  currencySymbolAlternates = roubleAlternateKeys
+  spaceBar = "רווח"
+  language = "עברית"
+  invalidCommandMsg = "אין מידע"
+  baseAutosuggestions = ["אתמ", "אני", "היי"]
+  numericAutosuggestions = ["", "", ""]
+
+  translateKeyLbl = "Translate"
```

Comment:
We also similarly need these to be localized into Hebrew :) Same for Conjugate and Plural.